src/AzSK.ADO/Framework/Abstracts/MetaInfoProvider.ps1

```diff
@@ -0,0 +1,284 @@
+class MetaInfoProvider {
+    hidden static [MetaInfoProvider] $_instance = [MetaInfoProvider]::new()
+    static [MetaInfoProvider] $Instance = [MetaInfoProvider]::GetInstance()
+
+	hidden [bool] $bUseADOInfoAPI 
+	hidden [string] $FuncAPI = '/api/getadoinfo'; 
+	hidden [string] $code;
+    hidden [string] $baseURL;
+    hidden [PSObject] $ControlSettings; 
+    
+    hidden [PSObject] $buildSTDetails;
+    hidden [PSObject] $releaseSTDetails;
+    hidden [PSObject] $svcConnSTDetails;
+    hidden [PSObject] $agtPoolSTDetails;
+    hidden [PSObject] $varGroupSTDetails;
+    hidden [PSObject] $serviceTreeDetails;
+
+    hidden MetaInfoProvider() {
+    }
+
+    hidden static [MetaInfoProvider] GetInstance() {
+        return [MetaInfoProvider]::_instance
+    }
+
+    [bool] CheckIsADOInfoAPIEnabled()
+	{
+        if ($null -eq $this.ControlSettings)
+        {
+            $this.ControlSettings = [ConfigurationManager]::LoadServerConfigFile("ControlSettings.json");
+        }
+        $adoInfoAPI = $this.ControlSettings.ADOInfoAPI;
+        if ($null -ne $adoInfoAPI -and $Env:AzSKADODoNotUseADOInfoAPI -ne $true)
+        {
+            #TODO
+            #$adoInfoAPI.Enabled = $false;
+            if ($adoInfoAPI.Enabled)
+            {
+                $this.bUseADOInfoAPI = $true;
+                $this.code = $adoInfoAPI.Code;
+                $this.baseURL = $adoInfoAPI.Endpoint;
+            }
+		}
+		return $this.bUseADOInfoAPI;
+	}
+
+	[PSObject] CallADOInfoAPI($queryString )
+	{
+        $adoInfoInvokeURL = $this.baseURL + $this.FuncAPI + $queryString
+        $Header = @{
+            "x-functions-key" = $this.code;
+        }
+        $rsrcList = $null
+        
+        try 
+        {
+            $rsrcList = Invoke-RestMethod -Method 'GET' -Uri $adoInfoInvokeURL -Headers $Header
+        }
+        catch
+        {
+            Write-Host "Error calling ADO Info API. `r`nPlease contact your project's ADO security team." -ForegroundColor Red 
+        }
+        return $rsrcList;
+	}
+
+	[void] FetchMappingFiles($ResourceTypeName)
```

Comment:
rearrange the sequence of methods in this class

Reply:
rearraanged